pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -234,7 +235,7 @@ func (m *kubeGenericRuntimeManager) calculateLinuxResources(cpuRequest, cpuLimit
 	}
 
 	// runc requires cgroupv2 for unified mode
-	if isCgroup2UnifiedMode() {
+	if isCgroup2UnifiedMode() && !m.singleProcessOOMKill {
```

Comment:
You should add some unit tests for this: https://github.com/kubernetes/kubernetes/blob/ee5e6e899b100edbd64860bc20337e11711e2670/pkg/kubelet/kuberuntime/kuberuntime_container_linux_test.go#L238

Test are all ready there so let's at least cover this case.